### ISSUE TITLE
fix(model): apply configured causal windows in V4.1 plain attention

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13303]
# fix(model): apply configured causal windows in V4.1 plain attention

```routing
repo: fak
lane: model
paths: ["internal/model/v41_forward.go","internal/model/v41_plain_window_mask.go","internal/model/v41_plain_window_mask_test.go"]
expected_steps: 3
```

## Working spine
Establish the full-prefix reference semantics required before bounded retained decode state can be trusted.

## Current state
V4.1 full attention contracts every causal row in `v41_forward.go:2134-2145`, while parsed per-layer windows live in `Config.Window` and `windowForLayer`; GGUF maps `attention.sliding_window` in `internal/ggufload/deepseek41.go:104`. No exact open issue covers this V4.1 reuse seam.

## Why this is next
Leaves 06 and 07 cannot claim bounded local-layer retention while their full-prefix oracle ignores the configured sliding window.

## Parent context
anthony-chaudhary/fak#12640; performance frontier anthony-chaudhary/fak#13294.

## Core through-line
For each uncompressed plain layer, derive the allowed causal key interval from that layer's `Config.Window`/`windowForLayer` value and the absolute query position. A positive window `w` includes exactly causal keys `max(0,q-w+1)..q`; `-1` preserves full causal attention. Feed only those ordered rows to the existing score/softmax/value contraction. Preserve historical full-causal behavior until this leaf passes an independent scalar oracle.

## Gold-plating boundary
No compressed-role masking, retained decode state, kernel fusion, configuration reinterpretation or production incremental switch.

## Worker packet
Target worker: DS4.1 or Qwen3.8 capability level. Own only the listed paths. Three steps: write an independent scalar-mask regression without calling the helper under test; implement the row-selection seam; run focused/package gates, then land. No implementation prerequisite. Leaf 13 must land before 06. If scope exceeds three implementation files, return a narrower successor ticket.

## Done condition / witness
Witness: `go test ./internal/model -run 'TestV41PlainWindowMask' -count=1`
Confirm `go test -list` includes the named test; an empty match is not a pass. Run owning-package and boundary gates after the last edit.

## Verifiable Witness
```
go test ./internal/model -run 'TestV41PlainWindowMask' -count=1
```

## Definition of done
- [ ] Independent scalar-mask tests assert exact attended absolute row IDs and outputs for empty/short prefixes and query positions 0, 1, 127, 128 and 129.
- [ ] Positive local windows and `-1` global layers consume the correct `Config.Window` entry for each layer; the oracle duplicates the math directly and never calls production mask helpers.
- [ ] Focused and owning-package gates pass; an independent reviewer reads the configured production call path.

## Acceptance gate
All checklist items plus exact witness exit 0. Software validation is [SW-VERIFIED] only.

## Closure binding
Commit with DCO and this issue reference, explicit-path staging, and the (fak model) trailer; no agent-created PR.

## Likely files
- internal/model/v41_forward.go
- internal/model/v41_plain_window_mask.go
- internal/model/v41_plain_window_mask_test.go

## Expected steps
3

## Value
- Centrality: Core
- P1 Context: advanced - reference semantics precede retained-state optimization
- P2 Net value: preserved - causal parity is independently checked
- P3 Adaptation: N/A - no adaptive policy in this leaf
- P4 Operations: advanced - exact worker scope and witnessed call path

## Work estimate
Estimate: 2 points

## Overall completion contribution
Contribution: 2/8 points

## Completion standard
development

## Qualification status
This correctness leaf has no measured physical result; mission throughput remains outstanding.

**Comment by anthony-chaudhary** [comment]
**LANDED on `origin/main` at `b51eb102f` and independently witnessed. Closing mission leaf 13 (TICKET-13-plain-window-mask).**

Plan leaf 13 (parent epic #12640; perf frontier #13294). Unblocks leaf 06 (`fak#13305`).

## What changed (3 paths, `internal/model`)
- `internal/model/v41_plain_window_mask.go` (new): `v41PlainWindowKeys(q, w)` returns the ordered absolute causal key rows a plain (uncompressed) layer attends — positive window `w` selects exactly `max(0,q-w+1)..q`; the `-1` sentinel (and nil/short `Window`) keeps the historical full-causal prefix `0..q`. `v41PlainWindowIndexList` renders the positional index list the existing sink consumes for the flattened selected rows; `idx` is positional over the supplied KV array (`v41_sparse_attention.go` validates each against `opt.N`), so absolute IDs drive SELECTION and the sink addresses them `0..rows-1`.
- `internal/model/v41_forward.go`: the plain per-position branch now derives its row set from `cfg.windowForLayer(l)` instead of unconditionally using the full `[0,t]` prefix. Only WHICH ordered rows are contracted changes; the score/softmax/value math is untouched.
- `internal/model/v41_plain_window_mask_test.go` (new): the witness.

## Why
The plain (uncompressed) sink contraction ignored the configured window entirely, so every uncompressed layer attended globally even when the parsed checkpoint declares a sliding window. Leaves 06/07 cannot claim bounded local-layer retention while their full-prefix oracle ignores the configured window.

## Witness
```
go test ./internal/model -run 'TestV41PlainWindowMask' -count=1
ok  github.com/anthony-chaudhary/fak/internal/model  0.117s   (exit 0)
```
- `TestV41PlainWindowMask`: independent scalar oracle over row IDs at `q=0,1,127,128,129` for global/-1, wider-than-prefix, window 128, window 1, and the zero (global) case; the oracle duplicates the visibility math directly and never calls the production helpers.
- `TestV41PlainWindowMaskSinkOutput`: the existing sink contraction over window-selected rows reproduces a scalar softmax oracle over exactly those rows (global matches full causal; window 3 at q=5 matches rows 3..5).
- `TestV41PlainWindowMaskProductionPath`: drives the REAL `forwardV41` on the reduced fixture — an explicit `-1` window is full causal and `window=1` produces different, finite logits, proving the configured window reaches the production attention path (presence ≠ invokability).

Owning-package gate: `go test ./internal/model -count=1` => ok (227.8s). `gofmt` clean, `go vet` clean, `fak-boundary check --all` => PASS (0 violations across all 5 gates).

[SW-VERIFIED] only. Physical 100/30 throughput and coherent two-prompt generation remain with the #13294 / fak-private#1395 physical owner.

Closes #13303. Ref #12640, #13294.
